src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -644,54 +711,106 @@ def __add_multiple_exception_errors(self):
+        self.assertEqual(installation_msg['excludedPatchCount'], truncated_substatus_msg['excludedPatchCount'])
+        self.assertEqual(installation_msg['pendingPatchCount'], truncated_substatus_msg['pendingPatchCount'])
+        self.assertEqual(installation_msg['installedPatchCount'], truncated_substatus_msg['installedPatchCount'])
+        self.assertEqual(installation_msg['failedPatchCount'], truncated_substatus_msg['failedPatchCount'])
+        self.assertEqual(installation_msg['startTime'], truncated_substatus_msg['startTime'])
+        self.assertEqual(installation_msg['lastModifiedTime'], truncated_substatus_msg['lastModifiedTime'])
+        self.assertEqual(installation_msg['maintenanceRunId'], truncated_substatus_msg['maintenanceRunId'])
+
+    def __assert_assessment_tombstone_record(self, truncated_substatus_file_data, tombstone_count=1):
+        truncated_substatus_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        self.assertEqual(truncated_substatus_msg['patches'][-1]['patchId'], "Truncated_patch_list_id")
+        self.assertEqual(truncated_substatus_msg['patches'][-1]['classifications'], ['Other'])
+        self.assertTrue("additional updates of classification" in truncated_substatus_msg['patches'][-1]['name'][0])
+
+        if tombstone_count >= 2:
+            self.assertEqual(truncated_substatus_msg['patches'][-2]['patchId'], "Truncated_patch_list_id")
+            self.assertEqual(truncated_substatus_msg['patches'][-2]['classifications'], ['Security'])
+
+        if tombstone_count >= 3:
+            self.assertEqual(truncated_substatus_msg['patches'][-3]['patchId'], "Truncated_patch_list_id")
+            self.assertEqual(truncated_substatus_msg['patches'][-3]['classifications'], ['Critical'])
+
+    def __write_assessment_to_status_file(self, config_operation, patch_count_assessment, patch_count_security=0, patch_count_other=0, status='transitioning', classification='Critical'):
         self.runtime.execution_config.operation = config_operation
         self.runtime.status_handler.set_current_operation(config_operation)
 
-        if run == 'assessment':
-            test_patches, test_patches_version = self.__set_up_patches_func(patch_count)
-            self.runtime.status_handler.set_package_assessment_status(test_patches, test_patches_version, classification=classification, status=package_status)
+        # random_char to ensure the patches are unique due to __set_up_packages_func remove duplicates
+        self.__run_assessment_package_set_up(patch_count_other, classification='Other', random_char='b')  # Other assessment patches
+        self.__run_assessment_package_set_up(patch_count_security, classification='Security', random_char='a')  # Security assessment patches
+        self.__run_assessment_package_set_up(patch_count_assessment, classification=classification)
```

Comment:
I don't see this solving the issue we discussed in call today. It is still following the same lines where one classification is passed in as input and others are assumed. 

Does not make the function reusable as it should be. What if classification coming in as function input is 'Security'?

Reply:
new changes:
![image](https://github.com/Azure/LinuxPatchExtension/assets/127874208/0fae2220-8b21-4b8c-93e1-0a7163af45c0)